server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileChangelogAction.java

```diff
@@ -0,0 +1,142 @@
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.qualityprofile.ws;
+
+import org.sonar.api.resources.Languages;
+import org.sonar.api.server.ws.*;
+import org.sonar.api.server.ws.WebService.NewAction;
+import org.sonar.api.server.ws.WebService.NewController;
+import org.sonar.api.server.ws.WebService.Param;
+import org.sonar.api.utils.DateUtils;
+import org.sonar.api.utils.Paging;
+import org.sonar.api.utils.text.JsonWriter;
+import org.sonar.core.persistence.DbSession;
+import org.sonar.server.db.DbClient;
+import org.sonar.server.es.SearchOptions;
+import org.sonar.server.exceptions.NotFoundException;
+import org.sonar.server.qualityprofile.QProfileActivity;
+import org.sonar.server.qualityprofile.QProfileActivityQuery;
+import org.sonar.server.qualityprofile.QProfileFactory;
+import org.sonar.server.qualityprofile.QProfileService;
+import org.sonar.server.search.Result;
+
+import java.util.Date;
+import java.util.Map.Entry;
+
+public class QProfileChangelogAction implements BaseQProfileWsAction {
+
+  private static final String PARAM_SINCE = "since";
+  private static final String PARAM_TO = "to";
+
+  private QProfileService service;
+  private DbClient dbClient;
+  private QProfileFactory profileFactory;
+  private Languages languages;
+
+  public QProfileChangelogAction(QProfileService service, DbClient dbClient, QProfileFactory profileFactory, Languages languages) {
+    this.service = service;
+    this.dbClient = dbClient;
+    this.profileFactory = profileFactory;
+    this.languages = languages;
+  }
+
+  @Override
+  public void define(NewController context) {
+    NewAction changelog = context.createAction("changelog")
+      .setSince("5.2")
+      .setDescription("Get the history of changes on a quality profile: rule activation/deactivation, change in parameters/severity. " +
+        "Events are ordered by date in descending order (most recent first).")
+      .setHandler(this)
+      .setResponseExample(getClass().getResource("example-changelog.json"));
+
+    QProfileIdentificationParamUtils.defineProfileParams(changelog, languages);
+
+    changelog.addPagingParams(50);
+
+    changelog.createParam(PARAM_SINCE)
+      .setDescription("Start date for the changelog.")
+      .setExampleValue("2011-04-25");
```

Comment:
Reading the Rails code, it seemed that it was only date, but looking at [this thread on the user ML](http://sonarqube.15.x6.nabble.com/Link-to-quality-profiles-changes-doesn-t-work-in-the-Events-widget-tt5033551.html), it seems that indeed datetimes are allowed...